### PR TITLE
[8.0][event_training] Training module

### PR DESCRIPTION
**Pull request by yajo**
This module extends events to support training events and courses.
- Source: Refactorization of #14
- Design: [Google Docs](https://docs.google.com/document/d/1WiOB1yAWTR2aROpM6O632QWXwgG4hLRUpTHU2Pv4yEQ/edit?usp=sharing)
- Target: https://github.com/OCA/l10n-spain/issues/174
- Dependencies: #21 (bots will fail until merged)

Rebased, squashed all except one real commit that I hope will help to prevent the exception flamewar. Please avoid it in this thread, I'm tired of it.

### CODE REVIEW

**Comment by rafaelbn** [comment]
Please rebase

**Comment by yajo** [comment]
Done, ready to review.

File: event_training/models/product.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# © 2015 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis
+
+from openerp import _, api, fields, models
+from .. import exceptions as ex
+
+
+class EventTrainingProductABC(models.AbstractModel):
+    _name = "event_training.product_abc"
```
Comment (dreispt) [inline event_training/models/product.py:10]: abc?? Not an expressive word I must say. 
How about "Mixin" or "Logic"?

File: event_training/views/diploma_delivery_receipt_report.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+     © 2015 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis -->
+
+<openerp>
+<data>
```
Comment (dreispt) [inline event_training/views/diploma_delivery_receipt_report.xml:6]: noupdate=1?

File: event_training/views/diploma_delivery_receipt_report.xml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+<!-- Templates -->
+<template id="diploma_receipt">
+    <t t-call="report.html_container">
+        <t t-foreach="doc_ids" t-as="doc_id">
+            <t t-raw="translate_doc(
```
Comment (dreispt) [inline event_training/views/diploma_delivery_receipt_report.xml:12]: yes!

File: event_training/views/diploma_report.xml [hunk 4]
```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+     © 2015 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis -->
+
+<openerp>
+<data>
```
Comment (dreispt) [inline event_training/views/diploma_report.xml:6]: noupdate=1?

**Comment by dreispt** [comment]
Other than a couple of minor notes, :+1:

File: event_training/models/product.py (hunk 1 above)
Comment (yajo) [inline event_training/models/product.py:10]: It comes from [Python's abc module](https://docs.python.org/2.7/library/abc.html).

File: event_training/views/diploma_delivery_receipt_report.xml (hunk 3 above)
Comment (yajo) [inline event_training/views/diploma_delivery_receipt_report.xml:12]: pardon? :sweat_smile:

File: event_training/views/diploma_report.xml (hunk 4 above)
Comment (yajo) [inline event_training/views/diploma_report.xml:6]: https://github.com/OCA/event/pull/23#discussion_r53134675

File: event_training/views/diploma_delivery_receipt_report.xml (hunk 2 above)
Comment (yajo) [inline event_training/views/diploma_delivery_receipt_report.xml:6]: https://github.com/OCA/event/pull/23#discussion_r53134675

File: event_training/models/product.py (hunk 1 above)
Comment (dreispt) [inline event_training/models/product.py:10]: I see. But that's similar to use "Model" to name a Model class (`class EventModel`) - it's not specific, and arguably redundant.
Consider replacing it with something like "Mixin", "Logic", "Common" or similar expression.

**Pull request closed without merging**